lib/LocalizationLoader.js

```diff
@@ -511,74 +425,70 @@ LocalizationLoader.prototype._loadLocalizationsFromPath =
-		this._cookieConfig.name.length === 0) {
-		this._cookieConfig.name = DEFAULT_LOCALE_COOKIE_KEY;
-	}
-	if (typeof (this._cookieConfig.maxAge) !== 'number') {
-		this._cookieConfig.maxAge = DEFAULT_LOCALE_COOKIE_MAX_AGE;
+			.catch(this._logger.error);
 	}
 
-	if (typeof (this._cookieConfig.path) !== 'string' ||
-		this._cookieConfig.path.length === 0) {
-		this._cookieConfig.path = LOCALE_COOKIE_PATH;
+	/**
+	 * Prepares cookie configuration.
+	 * @param {Object} config Cookie configuration.
+	 * @param {string?} config.name Cookie name ('locale' by default).
+	 * @param {string?} config.maxAge Cookie max-age
+	 * (3 155 692 600 secs, 100 years by default).
+	 * @param {string?} config.domain Cookie domain.
+	 * @param {string?} config.path Cookie path.
+	 * @private
+	 */
+	_prepareCookieConfig(config) {
+		this._cookieConfig = Object.create(config);
+		if (typeof (this._cookieConfig.name) !== 'string' ||
+			this._cookieConfig.name.length === 0) {
+			this._cookieConfig.name = DEFAULT_LOCALE_COOKIE_KEY;
+		}
+		if (typeof (this._cookieConfig.maxAge) !== 'number') {
+			this._cookieConfig.maxAge = DEFAULT_LOCALE_COOKIE_MAX_AGE;
+		}
+
+		if (typeof (this._cookieConfig.path) !== 'string' ||
+			this._cookieConfig.path.length === 0) {
+			this._cookieConfig.path = LOCALE_COOKIE_PATH;
+		}
 	}
-};
 
-/**
- * Sets locale to HTTP response cookie.
- * @param {http.IncomingMessage} request HTTP request.
- * @param {http.ServerResponse} response HTTP response.
- * @param {string} locale Locale name.
- */
-LocalizationLoader.prototype._setLocaleToResponse =
-	function (request, response, locale) {
+	/**
+	 * Sets locale to HTTP response cookie.
+	 * @param {http.IncomingMessage} request HTTP request.
+	 * @param {http.ServerResponse} response HTTP response.
+	 * @param {string} locale Locale name.
+	 */
+	_setLocaleToResponse(request, response, locale) {
 		// expire date = current date + max-age in seconds
-		var expireDate = new Date((new Date()).getTime() +
-			this._cookieConfig.maxAge * 1000),
-			localeSetup = this._cookieConfig.name + '=' + locale,
-			cookieSetup = localeSetup;
+		const expireDate = new Date((new Date()).getTime() +
+				this._cookieConfig.maxAge * 1000);
+		const localeSetup = `${this._cookieConfig.name}=${locale}`;
+		let cookieSetup = localeSetup;
 
 		// http://tools.ietf.org/html/rfc6265#section-4.1.1
-		cookieSetup += '; Max-Age=' + this._cookieConfig.maxAge;
-		cookieSetup += '; Expires=' + expireDate.toUTCString();
+		cookieSetup += `; Max-Age=${this._cookieConfig.maxAge}`;
+		cookieSetup += `; Expires=${expireDate.toUTCString()}`;
 
 		if (this._cookieConfig.path) {
-			cookieSetup += '; Path=' + String(this._cookieConfig.path || '');
+			cookieSetup += `; Path=${String(this._cookieConfig.path || '')}`;
 		}
 
 		if (this._cookieConfig.domain) {
-			cookieSetup +=
-				'; Domain=' + String(this._cookieConfig.domain || '');
+			cookieSetup += `; Domain=${String(this._cookieConfig.domain || '')}`;
 		}
```

Comment:
Please check if it's up to date with https://github.com/catberry/catberry/blob/develop/lib/base/CookieWrapperBase.js#L86-L111